code/drasil-lang/lib/Language/Drasil/CodeExpr/Extract.hs

```diff
@@ -90,7 +91,7 @@ eNamesRI' (UpFrom il)     = eNames' (snd il)
 
 -- | Get dependencies from an equation.
 eDep :: CodeExpr -> [UID]
-eDep = nub . eNames
+eDep = toList . fromList . eNames
```

Comment:
I'm fairly certain that `toList . fromList` would also sort the list, since Haskell's internal `Set` is really an ordered binary tree (like we discussed yesterday, @NoahCardoso). That is to say, I'm a bit skeptical that `eDep` uses care for the order of the `UID`s. Have you tried switching the type here from a list to a set instead? Our `eDep`-like (UID-dependency searching) tools have a lot of `nub` usage, so switching to a set type might make sense.

Reply:
@balacij should I change every list of UIDs to Set of UIDs?

Reply:
I think there will be many, but I'm not sure if all should be changed. For example, if we have a list of UIDs in a chunk, we will need to check if order matters/makes sense.

Reply:
It's likely a lot more subtle than that: the question is when does order matter, and when it doesn't. And sometimes order matters only for display purposes.

We do over-use lists, but that's quite typical of most Haskell projects which make lists too easy to use!